protocol/x/clob/e2e/long_term_orders_test.go

```diff
@@ -233,3 +246,942 @@ func TestImmediateExecutionLongTermOrders(t *testing.T) {
+	require.True(t, found, "Order is not expired and should still be in state")
+
+	// block time ten, still not expired
+	ctx = tApp.AdvanceToBlock(3, testapp.AdvanceToBlockOptions{
+		BlockTime: time.Unix(10, 0).UTC(),
+	})
+	_, found = tApp.App.ClobKeeper.GetLongTermOrderPlacement(ctx, order.OrderId)
+	require.True(t, found, "Order is not expired and should still be in state")
+
+	// block time fifteen, expired
+	ctx = tApp.AdvanceToBlock(4, testapp.AdvanceToBlockOptions{
+		BlockTime: time.Unix(15, 0).UTC(),
+	})
+	_, found = tApp.App.ClobKeeper.GetLongTermOrderPlacement(ctx, order.OrderId)
+	require.False(t, found, "Order is expired and should not be in state")
+}
+
+func TestPlaceLongTermOrder(t *testing.T) {
+	tApp := testapp.NewTestAppBuilder().Build()
+	ctx := tApp.InitChain()
+
+	// subaccounts for indexer expectation assertions
+	aliceSubaccount := tApp.App.SubaccountsKeeper.GetSubaccount(ctx, constants.Alice_Num0)
+	bobSubaccount := tApp.App.SubaccountsKeeper.GetSubaccount(ctx, constants.Bob_Num0)
+
+	// order msgs
+	LongTermPlaceOrder_Alice_Num0_Id0_Clob0_Buy1_Price50000_GTBT5 := *clobtypes.NewMsgPlaceOrder(
+		clobtypes.Order{
+			OrderId: clobtypes.OrderId{
+				SubaccountId: *aliceSubaccount.Id,
+				ClientId:     0,
+				OrderFlags:   clobtypes.OrderIdFlags_LongTerm,
+				ClobPairId:   0,
+			},
+			Side:         clobtypes.Order_SIDE_BUY,
+			Quantums:     10_000_000_000, // 1 BTC, assuming atomic resolution of -10
+			Subticks:     500_000_000,    // 50k USDC / BTC, assuming QCE of -8
+			GoodTilOneof: &clobtypes.Order_GoodTilBlockTime{GoodTilBlockTime: 5},
+		},
+	)
+	LongTermPlaceOrder_Alice_Num0_Id0_Clob0_Buy2_Price50000_GTBT5 := *clobtypes.NewMsgPlaceOrder(
+		clobtypes.Order{
+			OrderId: clobtypes.OrderId{
+				SubaccountId: *aliceSubaccount.Id,
+				ClientId:     0,
+				OrderFlags:   clobtypes.OrderIdFlags_LongTerm,
+				ClobPairId:   0,
+			},
+			Side:         clobtypes.Order_SIDE_BUY,
+			Quantums:     20_000_000_000,
+			Subticks:     500_000_000,
+			GoodTilOneof: &clobtypes.Order_GoodTilBlockTime{GoodTilBlockTime: 5},
+		},
+	)
+	PlaceOrder_Bob_Num0_Id0_Clob0_Sell1_Price50000_GTB20 := *clobtypes.NewMsgPlaceOrder(
+		clobtypes.Order{
+			OrderId:      clobtypes.OrderId{SubaccountId: constants.Bob_Num0, ClientId: 0, ClobPairId: 0},
+			Side:         clobtypes.Order_SIDE_SELL,
+			Quantums:     10_000_000_000,
+			Subticks:     500_000_000,
+			GoodTilOneof: &clobtypes.Order_GoodTilBlock{GoodTilBlock: 20},
+		},
+	)
+	PlaceOrder_Bob_Num0_Id1_Clob0_Sell1_Price50000_GTB20 := *clobtypes.NewMsgPlaceOrder(
+		clobtypes.Order{
+			OrderId:      clobtypes.OrderId{SubaccountId: constants.Bob_Num0, ClientId: 1, ClobPairId: 0},
+			Side:         clobtypes.Order_SIDE_SELL,
+			Quantums:     10_000_000_000,
+			Subticks:     500_000_000,
+			GoodTilOneof: &clobtypes.Order_GoodTilBlock{GoodTilBlock: 20},
+		},
+	)
+	LongTermPlaceOrder_Alice_Num0_Id0_Clob0_Buy100_Price10_GTBT15_PO := *clobtypes.NewMsgPlaceOrder(
+		MustScaleOrder(
```

Comment:
Why is this scaled but the other ones aren't?

Reply:
Was just re-using a pre-existing order that existed in constants which needed to be scaled. For clarity I'll just create the order manually instead of calling the scale function

Reply:
Just re-using an existing post-only order which was created using the unscaled order format (where quantums should be multiplied by step_base_quantums and subticks should be multiplied by subticks per tick)